### ISSUE TITLE
Accessibility does not announce number of selected tabs in multi-select mode

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/fenix#19784]
### Steps to reproduce
1. Turn on accessibility mode.
2. Open the tabs tray.
3. Select multiple tabs.

### Expected
- Selecting a new tab should announce the new tab selection size.

### Actual
- No announcement is made.

**Comment by gabrielluong** [comment]
Some relevant strings and code context can be found https://github.com/mozilla-mobile/fenix/pull/19740#discussion_r644329795

**Comment by jonalmeida** [comment]
This is the old logic for accessiblity:
```kotlin
if (oldMode.selectedItems != state.mode.selectedItems) {
  val unselectedItems = oldMode.selectedItems - state.mode.selectedItems

  state.mode.selectedItems.union(unselectedItems).forEach { item ->
    if (view.context.settings().accessibilityServicesEnabled) {
      view.announceForAccessibility(
        if (unselectedItems.contains(item)) view.context.getString(
          R.string.tab_tray_item_unselected_multiselect_content_description, 
          item.title
        ) 
        else view.context.getString(
          R.string.tab_tray_item_selected_multiselect_content_description, 
          item.title
        )
      )
    }
    updateTabsForSelectionChanged(item.id)
  }
}
```

This is quite heavy when you're actually using accessibility since it announces all selected tabs from the start before it gets to your last added item.

I recommend we follow what the system apps do instead, like the Contacts app when you're in select mode and you select a new contact: 
 - Hovering over it announces "Not ticked, tickbox, [name]". 
 - Selecting it then announces, "Ticked".
